### PR TITLE
Add zstd compression content to performance tuning doc

### PR DESCRIPTION
**Pull request by akashsha1**
### Description
This change adds a brief overview of the zstd compression introduced in opensearch v2.9 to the docs page.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by akashsha1** [comment]
@sarthakaggarwal97 please take a look

File: _tuning-your-cluster/performance.md [hunk 1]
```diff
@@ -134,3 +134,7 @@ curl -XPOST "es-endpoint/index-name/type-name/_bulk?pretty&filter_path=-took,-it
 { "doc" : {"user" : "example"} }
 ```
 {% include copy.html %}
+
+## Compression codecs
+
+Starting with OpenSearch 2.9, new codecs for compression are made available: `zstd` and `zstd_no_dict`. You can optionally specify a compression level for these in the `index.codec.compression_level` setting with values in the range [1,6]. These provide improved indexing performance at the expense of a slightly larger index size. [Benchmark]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/#benchmarking) data shows zstd providing 7% better writes throughput, and zstd_no_dict providing 14% better throughput. You can find more details about compression on [Indexing codecs]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs).
```
Comment (sarthakaggarwal97) [inline _tuning-your-cluster/performance.md:140]: ```These provide improved indexing performance at the expense of a slightly larger index size.```
this is slightly incorrect, we can remove this line.

shows ```zstd``` providing 7% better writes throughput, and ```zstd_no_dict``` providing 14% better throughput alongside over 30% improvements in storage over current ```default``` codec.

you can say something like this, feel free to rephrase

**Comment by sarthakaggarwal97** [comment]
thanks @akashsha1 for the review, minor comment, please check!

File: _tuning-your-cluster/performance.md (hunk 1 above)
Comment (akashsha1) [reply _tuning-your-cluster/performance.md:140]: I can edit it. However, the https://opensearch.org/docs/latest/im-plugin/index-codecs/ does mention the storage tradeoff.

Comment (sarthakaggarwal97) [reply _tuning-your-cluster/performance.md]: the storage trade off mentioned is between ```zstd``` and ```zstd_no_dict```. The indices created with ```zstd_no_dict``` will have slightly larger indices than those created with ```zstd```
In your previous revision, it seemed like there is a trade off with respect to current lz4 (default codec)

Comment (sarthakaggarwal97) [reply _tuning-your-cluster/performance.md]: This LGTM now! Thanks @akashsha1

File: _tuning-your-cluster/performance.md [hunk 2]
```diff
@@ -134,3 +134,7 @@ curl -XPOST "es-endpoint/index-name/type-name/_bulk?pretty&filter_path=-took,-it
 { "doc" : {"user" : "example"} }
 ```
 {% include copy.html %}
+
+## Compression codecs
+
+Starting with OpenSearch 2.9, new codecs for compression are made available: `zstd` and `zstd_no_dict`. You can optionally specify a compression level for these in the `index.codec.compression_level` setting with values in the range [1,6]. [Benchmark]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/#benchmarking) data shows zstd providing 7% better writes throughput, and zstd_no_dict providing 14% better throughput, along with 30% improvements in storage over current `default` codec. You can find more details about compression on [Indexing codecs]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs).
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/performance.md:140]:
```suggestion
In OpenSearch 2.9 and later, there are two new codecs for compression: `zstd` and `zstd_no_dict`. You can optionally specify a compression level for these in the `index.codec.compression_level` setting with values in the [1, 6] range. [Benchmark]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/#benchmarking) data shows that `zstd` provides a 7% better write throughput and `zstd_no_dict` provides a 14% better throughput, along with 30% improvements in storage over the `default` codec. For more information about compression, see [Index codecs]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs).
```

**Review by kolchfa-aws — APPROVED** [review]
@akashsha1 Thank you! One rewording suggestion, otherwise LGTM.

File: _tuning-your-cluster/performance.md [hunk 3]
```diff
@@ -137,4 +137,4 @@ curl -XPOST "es-endpoint/index-name/type-name/_bulk?pretty&filter_path=-took,-it
...
\ No newline at end of file
+In OpenSearch 2.9 and later, there are two new codecs for compression: `zstd` and `zstd_no_dict`. You can optionally specify a compression level for these in the `index.codec.compression_level` setting with values in the [1, 6] range. [Benchmark]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/#benchmarking) data shows that `zstd` provides a 7% better write throughput and `zstd_no_dict` provides a 14% better throughput, along with 30% improvements in storage over the `default` codec. For more information about compression, see [Index codecs]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs).
```
Comment (kolchfa-aws) [inline _tuning-your-cluster/performance.md:140]:
```suggestion
In OpenSearch 2.9 and later, there are two new codecs for compression: `zstd` and `zstd_no_dict`. You can optionally specify a compression level for these in the `index.codec.compression_level` setting with values in the [1, 6] range. [Benchmark]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/#benchmarking) data shows that `zstd` provides a 7% better write throughput and `zstd_no_dict` provides a 14% better throughput, along with 30% improvements in storage over the `default` codec. For more information about compression, see [Index codecs]({{site.url}}{{site.baseurl}}/im-plugin/index-codecs/).
```

**Comment by kolchfa-aws** [comment]
@akashsha1 Could you please fix DCO? Then we'll be ready to merge.

**Comment by akashsha1** [comment]
> @akashsha1 Could you please fix DCO? Then we'll be ready to merge.

@kolchfa-aws / @sarthakaggarwal97 - I ran into issues with branch merging, and couldn't resolve the DCO. 
So I opened a new PR with the already reviewed content. Please take a look and sign-off.
https://github.com/opensearch-project/documentation-website/pull/4814
PR#4814 should be good to merge.

**Comment by kolchfa-aws** [comment]
Closing in favor of #4814

**Pull request closed without merging**